src/app/user/register/register.component.html

```diff
@@ -6,37 +6,50 @@
                 </mat-form-field>
     
-                <!-- Password input -->
-                <mat-form-field appearance="outline">
-                    <mat-label>Password</mat-label>
-                    <input #userPassword matInput [type]="hide ? 'password' : 'text'">
-                    <button mat-icon-button matSuffix (click)="hide = !hide" [attr.aria-label]="'Hide password'" [attr.aria-pressed]="hide">
-                        <mat-icon>{{hide ? 'visibility_off' : 'visibility'}}</mat-icon>
-                    </button>
-                </mat-form-field>
+                <div formGroupName="userPasswordsGroup">
+                    <!-- Password input -->
+                    <mat-form-field appearance="outline">
+                        <mat-label>Password</mat-label>
+                        <input [formControl]="userPasswordControl" matInput [type]="hidePasswords ? 'password' : 'text'">
+                        <button mat-icon-button matSuffix (click)="hidePasswords = !hidePasswords" [attr.aria-label]="'Hide password'" [attr.aria-pressed]="hidePasswords">
+                            <mat-icon>{{hidePasswords ? 'visibility_off' : 'visibility'}}</mat-icon>
+                        </button>
+                        <mat-error *ngIf="userPasswordControl.hasError('required')">Please enter a password</mat-error>
+                    </mat-form-field>
 
-                <mat-form-field appearance="outline">
-                    <mat-label>Confirm</mat-label>
-                    <input #userConfirmPassword matInput [type]="hide ? 'password' : 'text'">
-                    <button mat-icon-button matSuffix (click)="hide = !hide" [attr.aria-label]="'Hide password'" [attr.aria-pressed]="hide">
-                        <mat-icon>{{hide ? 'visibility_off' : 'visibility'}}</mat-icon>
-                    </button>
-                    <mat-hint>Use 8 or more characters with atleast one letter, number & symbol</mat-hint>
-                </mat-form-field>
+                    <mat-form-field appearance="outline">
+                        <mat-label>Confirm</mat-label>
+                        <input [formControl]="userPasswordConfirmControl" matInput [type]="hidePasswords ? 'password' : 'text'" [errorStateMatcher]="confirmPasswordsMatch">
+                        <button mat-icon-button matSuffix (click)="hidePasswords = !hidePasswords" [attr.aria-label]="'Hide password'" [attr.aria-pressed]="hidePasswords">
+                            <mat-icon>{{hidePasswords ? 'visibility_off' : 'visibility'}}</mat-icon>
+                        </button>
+                        <!-- <mat-error *ngIf="userPasswordConfirmControl.hasError('required')">Please enter a password</mat-error> -->
+                        <mat-error>{{errorMessage}}</mat-error>
+
+                        <mat-hint>Use 8 or more characters with atleast one letter, number & symbol</mat-hint>
```

Comment:
Yes the function to enforce this rule is implemented but I commented it out so we would be able to create test accounts for now and so that we don't have to remember long complex passwords. Do you think I should just go ahead and enforce that requirement?

Reply:
i was more referring to just making sure the error message matches with what the current rule enforcement is as well as the mat-hint with this comment, we don't have to enforce the requirement yet but we should still try to make sure that the mat-hint is consistent with the error message

Reply:
Sounds good, I'm about to commit and push and update, do you think we should enforce it yet? Now that I think about it we should and just use a dummy 'safe' password and write it down so we don't forget to activate it for our demonstration during class next week